ui/dashboardApp/layout/signin/index.tsx

```diff
@@ -226,7 +226,7 @@ function TiDBSignInForm({ successRoute, onClickAlternative }) {
             label={t('signin.form.username')}
             rules={[{ required: true }]}
           >
-            <Input onInput={clearErrorMsg} prefix={<UserOutlined />} disabled />
+            <Input onInput={clearErrorMsg} prefix={<UserOutlined />} />
```

Comment:
Before TiDB 5.4 or something maybe better to keep disabling this field to avoid confusing user, since in these versions user is just not possible to use other accounts, due to lacking the `DASHBOARD_` privilege.

Reply:
It looks like we have a long-term need for version comparison util.

Reply:
It can also be implemented by moving this logic to PD, e.g. Pass "allowNonRootLogin: false" in PD 4.0.x, 5.0.x, ...

Reply:
How about implementing it in an individual PR.

My thought:

1. Add `allow-non-root-login` option likes `experimental`  for dashboard 
2. If PD check allow non root login, it pass allow-non-root-login as true to start dashboard, else false. So we don't need to care about how PD implements the check logic.

Reply:
How would this `If PD check allow non root login` being implemented? I think there should always be a place hard coding what kind of versions are allowed to do non root login, right? As it is a product feature, not a user provided choice.

Reply:
@shhdgit see this discussion about the `non-root-login` option